logstash/check.py

```diff
@@ -0,0 +1,304 @@
+            tags=tags,
+            timeout=timeout,
+            url=url
+        )
+        return config
+
+    def _get_data(self, url, config, send_sc=True):
+        """ Hit a given URL and return the parsed json
+        """
+        auth = None
+
+        # Load SSL configuration, if available.
+        # ssl_verify can be a bool or a string (http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification)
+        if isinstance(config.ssl_verify, bool) or isinstance(config.ssl_verify, str):
+            verify = config.ssl_verify
+        else:
+            verify = None
+        if config.ssl_cert and config.ssl_key:
+            cert = (config.ssl_cert, config.ssl_key)
+        elif config.ssl_cert:
+            cert = config.ssl_cert
+        else:
+            cert = None
+
+        try:
+            resp = requests.get(
+                url,
+                timeout=config.timeout,
+                headers=headers(self.agentConfig),
+                auth=auth,
+                verify=verify,
+                cert=cert
+            )
+            resp.raise_for_status()
+        except Exception as e:
+            if send_sc:
+                self.service_check(
+                    self.SERVICE_CHECK_CONNECT_NAME,
+                    AgentCheck.CRITICAL,
+                    message="Error {0} when hitting {1}".format(e, url),
+                    tags=config.service_check_tags
+                )
+            raise
+
+        return resp.json()
+
+    def _get_logstash_version(self, config):
+        """ Get the running version of logstash.
+        """
+        try:
+            data = self._get_data(config.url, config, send_sc=False)
+            version = data['version']
+            version = map(int, version.split('.')[0:3])
+        except Exception as e:
+            self.warning(
+                "Error while trying to get Logstash version "
+                "from %s %s"
+                % (config.url, str(e))
+            )
+            version = [1, 0, 0]
+
+
+        self.service_metadata('version', version)
+        self.log.debug("Logstash version is %s" % version)
+        return version
+
+    def check(self, instance):
+        config = self.get_instance_config(instance)
+
+        version = self._get_logstash_version(config)
+
+        stats_metrics = dict(self.STATS_METRICS)
+        if version < [6, 0, 0]:
+            stats_metrics.update(self.PIPELINE_METRICS)
+            pipeline_input_metrics = self.PIPELINE_INPUTS_METRICS
+            pipeline_output_metrics = self.PIPELINE_OUTPUTS_METRICS
+            pipeline_filters_metrics = self.PIPELINE_FILTERS_METRICS
+
+        stats_url = urlparse.urljoin(config.url, '/_node/stats')
+        stats_data = self._get_data(stats_url, config)
+
+        self._process_stats_data(stats_data, stats_metrics, config)
+
+        if version < [6, 0, 0]:
+            self._process_pipeline_input_data(stats_data, pipeline_input_metrics, config)
```

Comment:
`_process_pipeline_input_data`, `_process_pipeline_output_data` and `_process_pipeline_filters_data` are almost the same function.

We could merge them by calling them with the right data (`['pipeline']['plugins'][name]`) and giving them the tag name.
Something like that:
```python
self._process_pipeline_data(
  stats_data['pipeline']['plugins'].get('inputs', []),
  pipeline_input_metrics,
  config,
  "input_name")
```